### PR TITLE
Feature/donation aggregation

### LINKED ISSUE #84: Allow delegating money from Donate button in Campaigns and Milestones
If you are a delegate and you have money to delegate, clicking on the Donate button (or GivETH) should offer you the option to delegate the DAC money.

### PR DESCRIPTION
**Pull request by vojtechsimetka**
Fixes #84

### CODE REVIEW

**Comment by vojtechsimetka** [comment]
Deploy preview for *giveth-dapp* processing.

Built with commit 439fcb8b2b81bd8baaec4615ea45bf861a00fe17

https://app.netlify.com/sites/giveth-dapp/deploys/5b7432963813f025edac0920

File: src/components/DelegateMultipleButton.jsx [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+    this.donationsObserver = feathersClient
+      .service('donations')
+      .watch({ listStrategy: 'always' })
+      .find(query)
+      .subscribe(
+        r => {
+          const delegations = r.data.map(d => new Donation(d));
+          let amount = utils.fromWei(
+            delegations
+              .reduce((sum, d) => sum.plus(new BigNumber(d.amountRemaining)), new BigNumber('0'))
```
Comment (perissology) [inline src/components/DelegateMultipleButton.jsx]: just an fyi, but if you use utils.toBN(..) instead of new BigNumber, you can pass the BN object to fromWei & don't have to call .toString()

File: src/contextProviders/DonationProvider.jsx [hunk 2]
```diff
@@ -56,6 +56,7 @@ class DonationProvider extends Component {
             query: {
               giverAddress: this.props.currentUser.address,
               amountRemaining: { $ne: 0 },
+              pendingAmountRemaining: { $ne: 0 },
```
Comment (perissology) [inline src/contextProviders/DonationProvider.jsx]: why was this added? Doesn't this do an AND query

File: src/models/Campaign.js [hunk 3]
```diff
@@ -56,6 +57,10 @@ class Campaign extends BasicModel {
     return campaign;
   }
 
+  get type() {
+    return this._type;
+  }
+
```
Comment (perissology) [inline src/models/Campaign.js]: there is already a static getter for type. Does that not work for instances? If not, we should at least delegate the call to `Campaign.type()` instead of adding `_type` to the instance

File: src/models/DAC.js [hunk 4]
```diff
@@ -62,6 +63,10 @@ class DAC extends BasicModel {
     }
   }
 
+  get type() {
+    return this._type;
+  }
+
```
Comment (perissology) [inline src/models/DAC.js]: same comment as campaign

File: src/services/DonationService.js [hunk 5]
```diff
@@ -8,6 +9,7 @@ import { feathersClient } from '../lib/feathersClient';
 import { getWeb3 } from '../lib/blockchain/getWeb3';
 
 import ErrorPopup from '../components/ErrorPopup';
+// import Campaign from '../models/Campaign';
```
Comment (perissology) [inline src/services/DonationService.js]: remove please

File: src/contextProviders/DonationProvider.jsx (hunk 2 above)
Comment (vojtechsimetka) [reply src/contextProviders/DonationProvider.jsx]: I have a question here. This amount is the new `amountRemaining` once confirmed, right? In that case IMHO this is fine because both of these values need to be greater than 0, right? If anything I should actually be using the `pendingAmountRemaining` all the time rather than `amountRemaining`.
